Makefile

```diff
@@ -163,8 +163,38 @@ local-setup: docker-build setup-kind-cluster helm-manifest-install ## setup and
 	kubectl apply -f ./hack/localsetup/e2e-providerspc.yaml
 	kubectl apply -f ./hack/localsetup/e2e-secret-sync.yaml
 
+.PHONY: generate-kind-config
+generate-kind-config:
+	@echo "Generating kind-config.yaml based on Kubernetes version $(KIND_NODE_IMAGE_VERSION)"
+	@K8S_VERSION=$(KIND_NODE_IMAGE_VERSION); \
+	FEATURE_GATES=""; \
+	RUNTIME_CONFIG=""; \
+	if echo "$$K8S_VERSION" | grep -qE "^v1\.(26|27)\."; then \
+		FEATURE_GATES="ValidatingAdmissionPolicy: true"; \
+		RUNTIME_CONFIG="admissionregistration.k8s.io/v1alpha1: true"; \
+	elif echo "$$K8S_VERSION" | grep -qE "^v1\.(28|29)\."; then \
+		FEATURE_GATES="ValidatingAdmissionPolicy: true"; \
+		RUNTIME_CONFIG="admissionregistration.k8s.io/v1beta1: true"; \
+	else \
+		FEATURE_GATES=""; \
+		RUNTIME_CONFIG="admissionregistration.k8s.io/v1beta1: true"; \
+	fi; \
+	mkdir -p hack/localsetup; \
+	{ \
+	echo "kind: Cluster"; \
+	echo "apiVersion: kind.x-k8s.io/v1alpha4"; \
+	if [ -n "$$FEATURE_GATES" ]; then \
+		echo "featureGates:"; \
+		echo "  $$FEATURE_GATES"; \
+	fi; \
+	if [ -n "$$RUNTIME_CONFIG" ]; then \
+		echo "runtimeConfig:"; \
+		echo "  $$RUNTIME_CONFIG"; \
+	fi; \
+	} > hack/localsetup/kind-config.yaml
+
 .PHONY: setup-kind-cluster
-setup-kind-cluster: 
+setup-kind-cluster: generate-kind-config
```

Comment:
This change has been merged in an earlier PR to unblock CI. Could you rebase the PR to only make the changes for version?